jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -621,8 +635,13 @@ else if (!isUnreservedPctEncodedOrSubDelim(c))
                         case '/':
                             throw new IllegalArgumentException("No closing ']' for ipv6 in " + uri);
                         case ']':
-                            c = uri.charAt(++i);
-                            _host = uri.substring(mark, i);
+                            i++;
+                            String host = uri.substring(mark, i);
+                            URIUtil.validateInetAddress(host);
```

Comment:
Will this only occur during a detected IPLiteral only?  not a hostname right?
If it's a hostname, the resulting call to `InetAddress.getByName` will incur a DNS hit.

Reply:
Correct.   IPV6 state is only entered after seeing the corresponding `[` in HOST state.   The state machine here is not as refined as in 12, so there are probably still a few more strange cases here.... but I don't want to substantively change the behavior.